Detectors/TRD/calibration/src/T0Fit.cxx

```diff
@@ -0,0 +1,270 @@
+#include "Framework/TimingInfo.h"
+#include "Framework/InputRecord.h"
+#include "DataFormatsTRD/Constants.h"
+#include "TRDBase/GeometryBase.h"
+#include "TStopwatch.h"
+#include "CCDB/CcdbApi.h"
+#include "CCDB/BasicCCDBManager.h"
+#include "CommonUtils/NameConf.h"
+#include "CommonUtils/MemFileHelper.h"
+#include <TFile.h>
+#include <TTree.h>
+
+#include "TMath.h"
+#include "TCanvas.h"
+
+#include <string>
+#include <map>
+#include <memory>
+#include <ctime>
+
+using namespace o2::trd::constants;
+
+namespace o2::trd
+{
+//______________________________________________________________________________________________
+double ErfLandauChi2Functor::operator()(const double* par) const
+{
+  // provides chi2 estimate of PH profile comparing the y-value of profiles to
+  // par[0]*ROOT::Math::erf(x) + par[1]*TMath::Landau(x,par[2],par[3])
+  //
+  // par[0] : offset
+  // par[1] : amplitude
+  // par[2] : mean
+  // par[3] : sigma
+
+  double sum2 = 0;
+
+  for (int i = lowerBoundFit; i <= upperBoundFit; ++i) {
+
+    if (TMath::Abs(y[i]) < 1e-7) {
+      continue;
+    }
+
+    double funcVal = par[0] * TMath::Erf(x[i]) + par[1] * TMath::Landau(x[i], par[2], par[3]);
+
+    sum2 += TMath::Power(y[i] - funcVal, 2) / y[i];
+    // sum2 += TMath::Power((y[i] - funcVal)/yErr[i],2);
+  }
+
+  return sum2;
+}
+
+//______________________________________________________________________________________________
+using Slot = o2::calibration::TimeSlot<o2::trd::T0FitHistos>;
+
+void T0Fit::initOutput()
+{
+  // reset the CCDB output vectors
+  mInfoVector.clear();
+  mObjectVector.clear();
+}
+
+void T0Fit::initProcessing()
+{
+  if (mInitDone) {
+    return;
+  }
+
+  adcProfIncl = std::make_unique<TProfile>("adcProfIncl", "adcProfIncl", 30, -0.5, 29.5);
+  for (int iDet = 0; iDet < constants::MAXCHAMBER; ++iDet) {
+    adcProfDet[iDet] = std::make_unique<TProfile>(Form("adcProfDet_%i", iDet), Form("adcProfDet_%i", iDet), 30, -0.5, 29.5);
+  }
+
+  mFitFunctor.lowerBoundFit = 0;
+  mFitFunctor.upperBoundFit = 5;
+
+  double mParamsStart[4];
+  mParamsStart[0] = 100;
+  mParamsStart[1] = 500;
+  mParamsStart[2] = 0.5;
+  mParamsStart[3] = 0.5;
+
+  mFitter.SetFCN<ErfLandauChi2Functor>(4, mFitFunctor, mParamsStart);
+
+  ROOT::Math::MinimizerOptions opt;
+  opt.SetMinimizerType("Minuit2");
+  opt.SetMinimizerAlgorithm("Migrad");
+  opt.SetPrintLevel(0);
+  opt.SetMaxFunctionCalls(1000);
+  opt.SetTolerance(.001);
+  mFitter.Config().SetMinimizerOptions(opt);
+
+  mFuncErfLandau = std::make_unique<TF1>(
+    "fErfLandau", [&](double* x, double* par) { return par[0] * TMath::Erf(x[0]) + par[1] * TMath::Landau(x[0], par[2], par[3]); }, mFitFunctor.lowerBoundFit, mFitFunctor.upperBoundFit, 4);
+
+  // set tree addresses
+  if (mEnableOutput) {
+    mOutTree->Branch("t0_chambers", &t0_chambers);
+    mOutTree->Branch("t0_average", &t0_average);
+  }
+
+  mInitDone = true;
+}
+
+void T0Fit::retrievePrev(o2::framework::ProcessingContext& pc)
```

Comment:
I don't see where you are using the information from the previous CCDB object and since we want to only monitor T0 I don't think that is needed. I would simply remove this function and as you are already doing it assign a dummy value in case the fit is invalid